pkg/eventmonitor/config/config.go

```diff
@@ -0,0 +1,282 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package config
+
+import (
+	"fmt"
+
+	"os"
+
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/cmd/system-probe/config"
+	coreconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/ebpf"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	rsNS = "runtime_security_config"
+	evNS = "event_monitoring_config"
+)
+
+type Config struct {
+	ebpf.Config
+
+	// EnableKernelFilters defines if in-kernel filtering should be activated or not
+	EnableKernelFilters bool
+
+	// EnableApprovers defines if in-kernel approvers should be activated or not
+	EnableApprovers bool
+
+	// EnableDiscarders defines if in-kernel discarders should be activated or not
+	EnableDiscarders bool
+
+	// FlushDiscarderWindow defines the maximum time window for discarders removal.
+	// This is used during reload to avoid removing all the discarders at the same time.
+	FlushDiscarderWindow int
+
+	// SocketPath is the path to the socket that is used to communicate with the security agent and process agent
+	SocketPath string
+
+	// EventServerBurst defines the maximum burst of events that can be sent over the grpc server
+	EventServerBurst int
+
+	// PIDCacheSize is the size of the user space PID caches
+	PIDCacheSize int
+
+	// LoadControllerEventsCountThreshold defines the amount of events past which we will trigger the in-kernel circuit breaker
+	LoadControllerEventsCountThreshold int64
+
+	// LoadControllerDiscarderTimeout defines the amount of time discarders set by the load controller should last
+	LoadControllerDiscarderTimeout time.Duration
+
+	// LoadControllerControlPeriod defines the period at which the load controller will empty the user space counter used
+	// to evaluate the amount of events brought back to user space
+	LoadControllerControlPeriod time.Duration
+
+	// StatsTagsCardinality determines the cardinality level of the tags added to the exported metrics
+	StatsTagsCardinality string
+
+	// StatsdAddr defines the statsd address
+	StatsdAddr string
+
+	// AgentMonitoringEvents determines if the monitoring events of the agent should be sent to Datadog
+	AgentMonitoringEvents bool
```

Comment:
This parameter is used to define if the agent should send what we call `CustomEvents`. Right now these events are sent as logs, but they were primarily designed to trigger custom CWS events. I believe this field should be in the `security` package config for 2 reasons:
- unless I missed something, without the security-agent (and thus CWS enabled), this parameter won't have any effect.
- the goal of these events is primarily to notify customers of security coverage loss related to the agent itself by generating CWS signals. I don't think they are needed by any other module than CWS itself.